src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocParamOrderCheck.java

```diff
@@ -0,0 +1,334 @@
+ *    *&#47;
+ *   public static &lt;T, E, S&gt; int sampleMethod(T[] array, String str, E e ) {
+ *     return 2021;
+ *   }
+ *
+ *   &#47;**
+ *    * No violation for extra "t" parameter.
+ *    *
+ *    * &#64;param t desc          // OK, Extra parameter
+ *    *&#47;
+ *   public void sampleMethod2() {}
+ * }
+ * </pre>
+ * <p>
+ * Parent is {@code com.puppycrawl.tools.checkstyle.TreeWalker}
+ * </p>
+ * <p>
+ * Violation Message Keys:
+ * </p>
+ * <ul>
+  * <li>
+ * {@code javadoc.missed.html.close}
+ * </li>
+ * <li>
+ * {@code javadoc.parse.rule.error}
+ * </li>
+ * <li>
+ * {@code javadoc.tag.order}
+ * </li>
+ * <li>
+ * {@code javadoc.wrong.singleton.html.tag}
+ * </li>
+ * </ul>
+ *
+ * @since 8.42
+ */
+@StatelessCheck
+public class JavadocParamOrderCheck extends AbstractJavadocCheck {
+
+    /**
+     * A key is pointing to the warning message text in "messages.properties"
+     * file.
+     */
+    public static final String MSG_JAVADOC_TAG_ORDER = "javadoc.tag.order";
+
+    @Override
+    public int[] getDefaultJavadocTokens() {
+        return getRequiredJavadocTokens();
+    }
+
+    @Override
+    public int[] getRequiredJavadocTokens() {
+        return new int[] {
+            //JavadocTokenTypes.PARAMETER_NAME,
+            JavadocTokenTypes.PARAM_LITERAL,
+            JavadocTokenTypes.JAVADOC
+        };
+    }
+
+     @Override
+    public int[] getAcceptableJavadocTokens() {
+        return getRequiredJavadocTokens();
+    }
+
+    @Override
+    public void visitJavadocToken(DetailNode node) {
+
+        final DetailAST ast = getBlockCommentAst();
+        if (ast.getType() == TokenTypes.METHOD_DEF
+                 || ast.getType() == TokenTypes.CTOR_DEF) {
+            final List<DetailAST> methodParams = getMethodParameters(ast);
+            final List<DetailAST> methodTypeParams = CheckUtil.getTypeParameters(ast);
+
+            checkParamsOrder(node, methodParams, methodTypeParams);
+        }
+    }
+
+    /**
+     * Checks order of param tags for matching method parameters.
+     *
+     * @param tagNode descr.
+     * @param methodParams list of method parameters.
+     * @param methodTypeParams list of method type parameters.
+     */
+    private void checkParamsOrder(DetailNode tagNode,
+                                  final List<DetailAST> methodParams,
+                                  final List<DetailAST> methodTypeParams) {
+        methodTypeParams.addAll(methodParams);
+        final ListIterator<DetailAST> typeParamsIt = methodTypeParams.listIterator();
+        int listSize = methodTypeParams.size() - methodParams.size();
+        int count = 0;
+
+        DetailNode javadocTagSection =
+            JavadocUtil.findFirstToken(tagNode, JavadocTokenTypes.JAVADOC_TAG);
+
+        while (javadocTagSection != null && typeParamsIt.hasNext()) {
+            for (DetailNode child : javadocTagSection.getChildren()) {
+                if (child.getType() == JavadocTokenTypes.PARAMETER_NAME) {
```

Comment:
it seems correct to me but it's not working, 
> 
```
Expected :/home/aryaniiit002/Desktop/checkstyle/src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocparamorder/InputJavadocParamOrderIncorrect.java:15: Wrong order for 'p2' tag.
Actual   :Audit done.
```
it would be really helpful if anyone could tell me what's wrong, it's like logic is not picking javadoc param tags.

Reply:
made some changes in logic but it will still not work.

Reply:
help, somebody help.

Reply:
resolved